src/routes/stats/captable.js

```diff
@@ -22,157 +28,231 @@ const getAllStockClasses = async (issuerId) => {
-    // TODO: same here, verify
-    const votingPower = founderPreferredStockClasses.reduce((acc, stockClass) => {
-        const shares = totalStockIssuanceSharesByStockClass[stockClass._id] || 0;
-        return acc + Number(stockClass.votes_per_share) * shares;
+    const founderPreferredClasses = preferredStockClasses.filter(stockClass =>
+        founderIssuances.some(issuance => issuance.stock_class_id === stockClass._id)
+    );
+
+    const votingPower = founderIssuances.reduce((sum, issuance) => {
+        const stockClass = founderPreferredClasses.find(sc => sc._id === issuance.stock_class_id);
+        return sum + (stockClass ? stockClass.votes_per_share * Number(issuance.quantity) : 0);
     }, 0);
-    const votingPowerPercentage = totalVotingPower ? (votingPower / totalVotingPower).toFixed(2) : null;
 
     return {
-        initialSharesAuthorized,
         outstandingShares,
-        fullyDilutedShares,
-        fullyDilutedPercentage,
-        liquidationPreference,
+        sharesAuthorized: outstandingShares,
+        fullyDilutedShares: outstandingShares,
+        fullyDilutedPercentage: (outstandingShares / totalOutstandingShares * 100).toFixed(2),
+        liquidationPreference: Math.max(...founderPreferredClasses.map(sc => sc.liquidation_preference_multiple)),
         votingPower,
-        votingPowerPercentage,
+        votingPercentage: (votingPower / totalVotingPower * 100).toFixed(2)
     };
 };
 
+// modularizing the row creation for warrants and non-plan awards
+// for warrants, we need to use the exercise_triggers to get the quantity of shares that the warrant can convert to
+// for non-plan awards, we can just use the quantity
+const createWarrantAndNonPlanAwardsRow = (issuancesByStockClass, stockClasses, totalOutstandingShares, suffix, isWarrant = false) => {
+    return Object.entries(issuancesByStockClass).map(([stockClassId, issuances]) => {
+        const fullyDilutedShares = issuances.reduce((sum, issuance) => {
+            let quantity;
+            if (isWarrant) {
+                quantity = Number(issuance.exercise_triggers?.[0]?.conversion_right?.conversion_mechanism?.converts_to_quantity || 0);
+            } else {
+                quantity = Number(issuance.quantity);
+            }
+            return sum + (isNaN(quantity) ? 0 : quantity);
+        }, 0);
+
+        let name;
+        if (stockClassId === 'general') {
+            name = `General ${suffix}`;
+        } else {
+            const stockClass = stockClasses.find(sc => sc._id === stockClassId);
+            name = stockClass ? `${stockClass.name} ${suffix}` : `Unknown Stock Class ${suffix}`;
+        }
+
+        return {
+            name,
+            fullyDilutedShares,
+            fullyDilutedPercentage: (fullyDilutedShares / totalOutstandingShares * 100).toFixed(2)
+        };
+    });
+}
+
+const createEquityCompensationWithPlanAndTypeSummaryRows = (equityCompensationByStockPlanAndType, stockPlans, totalOutstandingShares) => {
```

Comment:
in `createEquityCompensationWithPlanAndTypeSummaryRows` the first argument is `equityCompensationByStockPlanAndType` it doesn't seem that we're giving the right argument in the function envokation